src/server/routes/api/document/index.js

```diff
@@ -1446,23 +1462,37 @@ http.get('/text/:id', function( req, res, next ){
     .catch( next );
 });
 
+const searchRelatedPapers = ( doc, interactionId ) => {
+  let article = doc.article();
+  if ( !article ) {
+    return [];
+  }
+
+  let templates;
+  if ( interactionId ) {
+    let intn = doc.get( interactionId );
+    templates = [ intn.toSearchTemplate() ];
+  }
+
+  templates = doc.toSearchTemplates();
+
+  let obj = { templates, article };
+  return indra.searchDocuments( obj );
+};
+
 http.get('/related-papers/:id', function( req, res, next ){
```

Comment:
Actually, in the previous issues our goal was to develop that route and the content of ``searchRelatedPapers()`` was in this route. In this PR I make a refactoring to extract ``searchRelatedPapers()`` out of it and use ``searchRelatedPapers()`` both in end point and for triggering related-papers work on submit.

I do not know if we will need it as a service in the future but I kept it since we wanted to have that service in the beginning. If there is no way of using it I can remove the service.